.github/workflows/build-and-test.yml

```diff
@@ -31,7 +31,12 @@ jobs:
         run:  dotnet build
         working-directory: ./skills/declarative/tests
 
-      # Run the tests
-      - name: Run tests
+      # Run the calendar tests
```

Comment:
While you're touching this, can you please update to only run the tests when needed? For example, I don't think there is any need to run tests if the change only touches generators for example.

Reply:
Sure, we can add a directory filter on the actions.